File: remote/src/main/java/com/kelsos/mbrc/repository/Repository.java
package com.kelsos.mbrc.repository;

import java.util.List;
import rx.Observable;

public interface Repository<T> {
  Observable<List<T>> getPageObservable(int offset, int limit);

  Observable<List<T>> getAllObservable();

  List<T> getPage(int offset, int limit);

  List<T> getAll();

  T getById(long id);

  void save(List<T> items);

  void save(T item);

  long count();
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/repository/Repository.kt
package com.kelsos.mbrc.repository

import rx.Observable

interface Repository<T> {
  fun getPageObservable(offset: Int, limit: Int): Observable<List<T>>

  val allObservable: Observable<List<T>>

  fun getPage(offset: Int, limit: Int): List<T>

  val all: List<T>

  fun getById(id: Long): T

  fun save(items: List<T>)

  fun save(item: T)

  fun count(): Long
}
